sound/soc/sof/intel/hda.c

```diff
@@ -383,6 +383,20 @@ static inline bool hda_sdw_check_wakeen_irq(struct snd_sof_dev *sdev)
 /* pre fw run operations */
 int hda_dsp_pre_fw_run(struct snd_sof_dev *sdev)
 {
+	if (sdev->first_boot) {
+		struct sof_intel_hda_dev *hdev = sdev->pdata->hw_pdata;
+
+		/* Check if IMR boot is usable */
+		if (!sof_debug_check_flag(SOF_DBG_IGNORE_D3_PERSISTENT) &&
+		    (sdev->fw_ready.flags & SOF_IPC_INFO_D3_PERSISTENT ||
```

Comment:
are the fw_ready flags set before the pre_fw_run callback? This would need to be commented with more details.

Reply:
aargh, you are right. The fw_ready.flags are coming from the FW_READY message _after_ the fw have booted.

If I want to keep the 'clever' way to keep the DMA buffer if IMR boot is not supported then we would only be able to store the buffer on the second boot. That could be fine, but it is a bit clumsy to document.

Probably dropping the cleverness and add a Kconfig instead to force this for products that would benefit.